lib/private/capabilitiesmanager.php

```
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @author Roeland Jago Douma <roeland@famdouma.nl>
+ *
+ * @copyright Copyright (c) 2015, ownCloud, Inc.
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OC;
+
+
+use OCP\Capabilities\ICapability;
+
+class CapabilitiesManager {
+
+	/**
+	 * @var \Closure[]
+	 */
+	private $capabilities = array();
+
+	/**
+	 * Get an array of al the capabilities that are registered at this manager
+     *
+	 * @return array
+	 */
+	public function getCapabilities() {
+		$capabilities = [];
+		foreach($this->capabilities as $capability) {
+			$c = $capability();
+			if ($c instanceof ICapability) {
+				$capabilities = array_replace_recursive($capabilities, $c->getCapabilities());
+			}
```

else log an error, maybe throw an exception even, so developers see why there stuff is failing.